pkg/placement/controller.go

```diff
@@ -413,13 +440,24 @@ func (c *Controller) reconcile(ctx context.Context, key util.Key) error {
 		if err := c.handleCRD(obj); err != nil {
 			return err
 		}
+	} else if util.IsPlacementDecision(obj) {
```

Comment:
Note that the if condition could be based on `key` rather than `obj`. If this `if` were moved to happen before the complex and buggy logic for workload objects then we would not the hack of an `isBeingCreated` field in the Key; syncing a reference to a PlacementDecision demonstrably needs no more than the Key. A work queue should only hold pure object references, not also instructions about an intended state change, because the relevant state change may be different by the time the object reference is dequeued for syncing.